### PR TITLE
feat(component): `<ExpandableArrow />`

### PR DESCRIPTION
**Pull request by alexcarpenter**
🔍 [Preview Link](https://react-components-git-acexpandable-arrow-hashicorp.vercel.app/components/expandablearrow)

---

## Description

Introduces `<ExpandableArrow />` as a component to be use across `<StandaloneLink />`, `<NextSteps />` and within `<Button />`.

### Example usage

```tsx
function Button() {
  const [isHovered, setIsHovered] = React.useState(false)
  return (
    <button
      onMouseOver={() => {
        setIsHovered(true)
      }}
      onMouseOut={() => {
        setIsHovered(false)
      }}
    >
      {children}
      <ExpandableArrow expanded={isHovered} />
    </button>
  )
}
```

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/expandable-arrow/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "@hashicorp/react-expandable-arrow",
+  "version": "0.0.1",
+  "description": "",
+  "author": "HashiCorp",
+  "license": "MPL-2.0",
+  "dependencies": {
+    "classnames": "^2.3.1"
+  },
+  "peerDependencies": {
+    "@hashicorp/mktg-global-styles": ">=3.x",
```
Comment (dstaley) [inline packages/expandable-arrow/package.json:11]: Is this needed? I don't see any usage of our global styles.

**Review by dstaley — APPROVED** [review]
LGTM!

**Review by pbortnick — APPROVED** [review]

**Review by nandereck — APPROVED** [review]
LGTM! 🚀

File: packages/expandable-arrow/package.json (hunk 1 above)
Comment (alexcarpenter) [reply packages/expandable-arrow/package.json]: 🔪 https://github.com/hashicorp/react-components/pull/798/commits/c32a22aa671a116264738a7090b2f312ef96d7e8

**Pull request merged**